ext/websocket/01_websocket.js

```diff
@@ -363,28 +364,16 @@
       } else if (this[_readyState] === OPEN) {
         this[_readyState] = CLOSING;
 
-        PromisePrototypeThen(
-          core.opAsync("op_ws_close", {
-            rid: this[_rid],
-            code,
-            reason,
-          }),
-          () => {
-            this[_readyState] = CLOSED;
-            const event = new CloseEvent("close", {
-              wasClean: true,
-              code: code ?? 1005,
-              reason,
-            });
-            this.dispatchEvent(event);
-            core.tryClose(this[_rid]);
-          },
-        );
+        core.opAsync("op_ws_close", {
```

Comment:
Need to probably add a `.catch(() => {})` to this so we don't end up with uncatchable promise rejections.

Reply:
maybe in that case we should close the ws and error with an error event?